lasagne/layers/merge.py

```diff
@@ -303,8 +303,9 @@ class ElemwiseMergeLayer(MergeLayer):
     Parameters
     ----------
     incomings : a list of :class:`Layer` instances or tuples
-        the layers feeding into this layer, or expected input shapes,
-        with all incoming shapes being equal
+        the layers feeding into this layer, or expected input shapes.
+        All input shapes must be equal or broadcastable. A dimension is
+        broadcastable if all non None dimensions are the same size or 1.
```

Comment:
I like the top one best. 

I'm not sure how we shoud handle 
input1 [32, 1]
input2 [32, 5]
input3 [32, None]

Should that give an error or just return None in the last dimensions? I think it's easier to debug if it raises an error?